### PR TITLE
Add API for UpDownCounter

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: api/src/main/java/io/opentelemetry/metrics/DoubleUpDownCounter.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
```
[the remark this answers is not in the document]
Comment (jkwatson) [reply api/src/main/java/io/opentelemetry/metrics/DoubleUpDownCounter.java:2]: yes, can you create an issue for it?

File: api/src/main/java/io/opentelemetry/metrics/DefaultMeter.java [hunk 2]
```diff
@@ -150,21 +174,31 @@ public DoubleCounter build() {
...
 
     @Override
     public NoopBoundLongCounter bind(String... labelKeyValuePairs) {
       Utils.validateLabelPairs(labelKeyValuePairs);
       return NoopBoundLongCounter.INSTANCE;
     }
 
+    // Similar to add but does not verify labels.
+    private static void verifyIncrement(long increment) {
+      Utils.checkArgument(increment >= 0, "Counters can only increase");
```
Comment (jkwatson) [inline api/src/main/java/io/opentelemetry/metrics/DefaultMeter.java:190]: This implementation is different from the one above. You could move it up out of the inner classes and put it one level higher so it doesn't need to be repeated.

File: api/src/main/java/io/opentelemetry/metrics/DoubleUpDownCounter.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jkwatson) [reply api/src/main/java/io/opentelemetry/metrics/DoubleUpDownCounter.java:2]: Note: we can also use a much more concise license header as suggested here:  https://github.com/open-telemetry/community/issues/113#issuecomment-619211462

[the remark this answers is not in the document]
Comment (thisthat) [reply api/src/main/java/io/opentelemetry/metrics/DoubleUpDownCounter.java:2]: Done in #1212

File: api/src/main/java/io/opentelemetry/metrics/DefaultMeter.java (hunk 2 above)
Comment (bogdandrutu) [reply api/src/main/java/io/opentelemetry/metrics/DefaultMeter.java:190]: It is different, because one uses long the other uses double.

Comment (jkwatson) [reply api/src/main/java/io/opentelemetry/metrics/DefaultMeter.java:190]: oh drat. stupid java.

File: api/src/main/java/io/opentelemetry/metrics/DoubleUpDownCounter.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (bogdandrutu) [reply api/src/main/java/io/opentelemetry/metrics/DoubleUpDownCounter.java]: Fixed the year, will fix other things when we fix the issue.

**Pull request merged**